### ISSUE TITLE
RulesCanBeReplacedWithComposition bug?

### ISSUE THREAD

**Issue by jiangzhen3s** [fsprojects/FSharpLint#140]
type ListTree<'a> = 
    | Node of 'a * ListTree<'a> list

let rec size (node: ListTree<'a>) = 
    match node with
    | Node(_, lst) -> List.fold (fun s -> size >> (+) s ) 1 lst

![qq 20151028011939](https://cloud.githubusercontent.com/assets/317738/10766322/fe614e40-7d11-11e5-8dff-703f959418c0.png)

**Comment by duckmatt** [comment]
Thanks for the report, I'll have a look into it

**Comment by jiangzhen3s** [comment]
thanks

**Comment by waf** [comment]
I ran into this today. If it's helpful, here's the simplest code I could find that triggered the warning:

```fsharp
open System

// Given a function of str -> str -> str
let concat a b = a + b
// Given a function of str -> int
let parse str = Int32.Parse(str)
// I want to call a function that takes a function of str -> str -> int
let doSomething (f: string -> string -> int) = f "1" "2"
// The linter says the function below could be replaced with function composition
let result = doSomething (fun s1 s2 -> concat s1 s2 |> parse)
```

**Comment by duckmatt** [comment]
@waf it is possible to create: `(fun s1 -> (concat s1) >> parse)` which is why I think it is complaining although the suggested code is harder to read so I think this rule will need to be updated to only be suggested in cases where it's clear that it would be of use

**Comment by duckmatt** [comment]
These have been fixed in [0.0.13](https://www.nuget.org/packages/FSharpLint.Core/) of the API
